### DISCUSSION TITLE
Add a distinced Backend for Tidb

### DISCUSSION THREAD

**Discussion by hawkingrei** [diesel-rs/diesel#2824] [Ideas]
Hello～ I'm a tidb developer from PingCAP.[TiDB](https://github.com/pingcap/tidb) is an open-source NewSQL database that aims for MySQL-compatibility. it is better to provide a TiDB dialect for the extended features compared with MySQL. so we want to do it in the diesel. Are you interested in completing on a mission?

PS: This is part of [pingcap/tidb#24194](https://github.com/pingcap/tidb/issues/24194)

**Comment by weiznich** [comment]
Thanks for reaching out with this. As stated in https://github.com/diesel-rs/diesel/issues/1997 we are currently not interested in having such a backend in tree, but we certainly open to support backends written in third party crates. See the following crates for examples:

* [diesel-oci](https://github.com/GiGainfosystems/diesel-oci)
* [rsfbclient](https://github.com/fernandobatels/rsfbclient)

On top of that the discussion in https://github.com/diesel-rs/diesel/discussions/2696 may be relevant to open up more API's for custom backend implementations.
